tests/ballerina-test-integration/src/test/java/org/ballerinalang/test/packaging/PackagingPushTestCase.java

```diff
@@ -0,0 +1,266 @@
+        Assert.assertEquals(response.getResponseCode(), 200, "Response code mismatched");
+
+    }
+
+    /**
+     * Get environment variables and add ballerina_home as a env variable the tmp directory.
+     *
+     * @return env directory variable array
+     */
+    private String[] getEnvVariables() {
+        List<String> variables = new ArrayList<>();
+
+        Map<String, String> envVarMap = System.getenv();
+        envVarMap.forEach((key, value) -> variables.add(key + "=" + value));
+        variables.add(ProjectDirConstants.HOME_REPO_ENV_KEY + "=" + tempHomeDirectory.toString());
+        variables.add("BALLERINA_DEV_STAGE_CENTRAL" + "=" + "true");
+
+        return variables.toArray(new String[0]);
+    }
+
+    /**
+     * Generate random package name.
+     *
+     * @param count number of characters required
+     * @return generated name
+     */
+    public String randomPackageName(int count) {
+        String upperCaseAlpha = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+        String lowerCaseAlpha = "abcdefghijklmnopqrstuvwxyz";
+        String alpha = upperCaseAlpha + lowerCaseAlpha;
+        StringBuilder builder = new StringBuilder();
+        while (count-- != 0) {
+            int character = (int) (Math.random() * alpha.length());
+            builder.append(alpha.charAt(character));
+        }
+        return builder.toString();
+    }
+
+    @AfterClass
+    private void cleanup() throws Exception {
+        ballerinaClient.stopServer();
+        Files.walk(tempHomeDirectory)
+             .sorted(Comparator.reverseOrder())
+             .forEach(path -> {
+                 try {
+                     Files.delete(path);
+                 } catch (IOException e) {
+                     Assert.fail(e.getMessage(), e);
+                 }
+             });
+        Files.walk(tempProjectDirectory)
+             .sorted(Comparator.reverseOrder())
+             .forEach(path -> {
+                 try {
+                     Files.delete(path);
+                 } catch (IOException e) {
+                     Assert.fail(e.getMessage(), e);
+                 }
+             });
+    }
+
+    /**
+     * Add file inside the src directory to the ZipOutputStream.
+     *
+     * @param zos      ZipOutputStream
+     * @param filePath file path of each file inside the driectory
+     * @throws IOException exception if an error occurrs when compressing
+     */
+    private static void addEntry(ZipOutputStream zos, Path filePath, String fileStr) throws IOException {
+        ZipEntry ze = new ZipEntry(fileStr);
+        zos.putNextEntry(ze);
+        Files.copy(filePath, zos);
+        zos.closeEntry();
+    }
+
+    /**
+     * Compresses files.
+     *
+     * @param outputStream outputstream
+     * @throws IOException exception if an error occurrs when compressing
+     */
+    private static void compressFiles(Path dir, OutputStream outputStream) throws IOException {
```

Comment:
this is special cased to a dir with two levels. it works but, it would be nicer to have a generic one like https://stackoverflow.com/a/32052016 (note liked code might have a bug on windows )

Reply:
Fixed with e85d0c1